server/util/uuid/uuid.go

```diff
@@ -156,6 +152,18 @@ func GetFailsafeHostID() string {
 	return failsafeID
 }
 
+func MigrateHostID(newConfigDir string) {
+	oldConfigDir, err := oldConfigDir()
+	if err != nil {
+		return
+	}
+	oldHostIDFilepath := path.Join(oldConfigDir, hostIDFilename)
+	newHostIDFilepath := path.Join(newConfigDir, hostIDFilename)
+	// This call will move an old-style host id file (if it exists) to the new
+	// host id file location.
+	os.Rename(oldHostIDFilepath, newHostIDFilepath)
```

Comment:
nit: use `disk.MoveFile`, since the files will likely be on different filesystems and `rename` could fail.

Also I think it's slightly safer to copy the file if it exists, rather than rename, in case for some reason the build root gets wiped but the home dir does not. This is a bit messy but IMO it's better to be overly cautious here to avoid wiping peoples' caches as much as we can